### PR TITLE
Exposed inverse function on matrix class.

### PR DESCRIPTION
**Pull request by ejoebstl**
Hello,

this adds an inverse function to the matrix class. It's WIP, I still need to test.

### CODE REVIEW

File: src/DlibDotNet.Native/dlib/matrix/matrix.h [hunk 1]
```diff
@@ -134,6 +134,12 @@ dlib::matrix<__TYPE__, __ROWS__, __COLUMNS__>& m = *(static_cast<dlib::matrix<__
 auto tmp = -m;\
 *ret = new dlib::matrix<__TYPE__, __ROWS__, __COLUMNS__>(tmp);
 
+#define matrix_operator_invert_template(__TYPE__, error, __ELEMENT_TYPE__, __ROWS__, __COLUMNS__, ...) \
+dlib::matrix<__TYPE__, __ROWS__, __COLUMNS__>& m = *(static_cast<dlib::matrix<__TYPE__, __ROWS__, __COLUMNS__>*>(matrix));\
+auto tmp = inv(m);\
```
Comment (takuya-takeuchi) [inline src/DlibDotNet.Native/dlib/matrix/matrix.h]: Could you add namespace to inv?
````
dlib::inv
````

### Reference
https://github.com/takuya-takeuchi/DlibDotNet/blob/47436a573c931fc9c9107442b10cf32524805378/src/DlibDotNet.Native/dlib/matrix/matrix_utilities.h#L128

**Comment by takuya-takeuchi** [comment]
What a wonderful work!!!
But I need you do a bit more work. (Yes, I know I sweat the details, Sorry.)
Please check review comment.

File: src/DlibDotNet.Native/dlib/matrix/matrix.h (hunk 1 above)
Comment (takuya-takeuchi) [reply src/DlibDotNet.Native/dlib/matrix/matrix.h]: Sorry, I forgot to request one more thing.
Could you use develop branch?

**Comment by ejoebstl** [comment]
I'm re creating this PR from a feature branch.

**Pull request closed without merging**